File: n8n-api/liderix_api/schemas/metrics.py
from __future__ import annotations

from datetime import datetime
from typing import Optional, List, Dict, Any
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class MetricDefinitionRead(BaseModel):
    id: UUID
    org_id: UUID
    name: str
    description: Optional[str] = None
    unit: Optional[str] = None
    formula: Optional[str] = None
    aggregation: Optional[str] = None
    meta_data: Optional[Dict[str, Any]] = None
    created_at: datetime
    updated_at: datetime
    deleted_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True, extra="forbid")


class MetricDefinitionListResponse(BaseModel):
    items: List[MetricDefinitionRead]
    total: int
    page: int
    page_size: int

    model_config = ConfigDict(from_attributes=True, extra="forbid")

File: apps/web-enterprise/src/lib/api/metrics.ts
import { api } from "@/lib/api/config"

export interface MetricDefinition {
  id: string
  org_id: string
  name: string
  description?: string | null
  unit?: string | null
  formula?: string | null
  aggregation?: string | null
  meta_data?: Record<string, unknown> | null
  created_at?: string
  updated_at?: string
  deleted_at?: string | null
}

interface MetricDefinitionListResponse {
  items: MetricDefinition[]
  total: number
  page: number
  page_size: number
}

export class MetricsAPI {
  static async listDefinitions(params?: { page?: number; page_size?: number; search?: string }): Promise<MetricDefinition[]> {
    const { data } = await api.get<MetricDefinitionListResponse>("/metrics/definitions", { params })
    return data.items || []
  }
}
